### PR TITLE
feat(WC2): Create Wallet Connect V2 Connector

### PR DESCRIPTION
**Pull request by fzavalia**
Add the WalletConnectV2Connector to be used when the WALLET_CONNECT_V2 provider type is provided.

## Important 

This PR installs `@walletconnect/ethereum-provider` which is compiled into JS that is incompatible with dapps using react-scripts v4. I suggest merging this PR when wallet connect v2 is absolutely needed. 

https://medium.com/walletconnect/weve-reset-the-clock-on-the-walletconnect-v1-0-shutdown-now-scheduled-for-june-28-2023-ead2d953b595

## Pre Releases

- https://github.com/decentraland/decentraland-connect/releases/tag/3.5.0-wc2-1

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -33,6 +67,27 @@ export function getConfiguration() {
   return configuration
 }
 
+/**
+ * Returns the list of chains that will be supported depending on the chainId provided.
+ * @param chainId The chain id.
+ * @returns A list of chains.
+ */
+export function getWalletConnectV2ConfigFromChainId(chainId: ChainId) {
```
Comment (LautaroPetaccio) [inline src/configuration.ts:75]: Would you mind testing this function?

File: src/connectors/WalletConnectV2Connector.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import { ConnectorUpdate } from '@web3-react/types'
+import EthereumProvider from '@walletconnect/ethereum-provider'
+import { ChainId } from '@dcl/schemas'
+import { AbstractConnector } from './AbstractConnector'
+
+export type WalletConnectV2ConnectorConfig = Parameters<
+  typeof EthereumProvider.init
+>[0]
+
+export class WalletConnectV2Connector extends AbstractConnector {
```
Comment (LautaroPetaccio) [inline src/connectors/WalletConnectV2Connector.ts:10]: The other connectors [seem to be tested](https://github.com/decentraland/decentraland-connect/blob/master/test/connectors.spec.ts) would you mind adding tests for this connector as well?

**Review by LautaroPetaccio — COMMENTED** [review]
Looks great, I just added some comments to the PR.

**Comment by fzavalia** [comment]
Closing in favor of #50

**Pull request closed without merging**